website/notifications/events/utils.py

```diff
@@ -0,0 +1,124 @@
+from website.notifications.emails import compile_subscriptions
+from website.notifications import utils, constants
+
+
+def get_file_subs_from_folder(addon, user, kind, path, name):
+    """Find the file tree under a specified folder."""
+    folder = dict(kind=kind, path=path, name=name)
+    file_tree = addon._get_file_tree(filenode=folder, user=user, version='latest-published')
+    files = []
+    list_of_files(file_tree, files)
```

Comment:
Try to avoid mutation of arguments, as it makes code harder to trace.

For example:

```python
def get_file_subs_from_folder(addon, user, kind, path, name):
   # ....
    files = list_of_files(file_tree)
    # ...


def list_of_files(file_object, files=None):
    files = files or []
     # ...
```